### DISCUSSION TITLE
Why there's no `additional_headers` setting for `jwt` auth?

### DISCUSSION THREAD

**Discussion by renom** [ory/oathkeeper#1181] [Q&A] [answered]
I need to add an additional header depending on auth method, e.g.:
1. If `jwt` method is used, add header `method=jwt`.
2. If `cookie_session` method is used, add header `method=cookie`.
3. If `bearer_token` method is used, add header `method=bearer`.

But setting `additional_headers` option is allowed only for `cookie_session` and `bearer_token` methods. Why?

**Comment by vinckr** [comment, marked as answer]
Hey @renom 
I think the reason why it's not available for the jwt method could be due to the nature of JWT itself. JWT are self-contained, meaning that all the necessary information should be contained within the token itself. This includes the method used for authentication, which is typically specified in the alg (algorithm) claim of the JWT. So here might not be a need to specify the method in an additional header.
If you need to add custom headers based on the authentication method, you might consider using Ory's `hydrator` mutator. This mutator allows for fetching additional data from external APIs, which can be then used by other mutators. It works by making an upstream HTTP call to an API specified in the Per-Rule Configuration section. The request is a POST request and it contains JSON representation of AuthenticationSession struct in body. As a response the mutator expects similar JSON object, but with extra or header fields modified. [Source](https://www.ory.sh/docs/oathkeeper/pipeline/mutator#hydrator)

**Reply by renom** [reply]
Hi @vinckr, thanks for the reply.
How can I detect what auth method is used from a `hydrator` app?
